src/basic/path-util.c

```diff
@@ -703,47 +721,64 @@ int find_executable_full(const char *name, const char *root, bool use_path_envva
                 if (r == 0)
                         break;
 
-                if (!path_is_absolute(element))
-                        continue;
+                if (find_executable_from_path(name, root, element, &last_error, ret_filename, ret_fd, fd))
+                        return 0;
+        }
 
-                if (!path_extend(&element, name))
-                        return -ENOMEM;
+        return last_error;
+}
 
-                if (root) {
-                        char *path_name;
+bool find_executable_from_path(
+        const char *name,
+        const char *root,
+        const char *element,
+        int *last_error_addr,
+        char **ret_filename,
+        int *ret_fd,
+        int fd) {
```

Comment:
This should be 16 space indent or aligned with the `(`

Reply:
`fd` doesn't need to be passed in. It can be local since the only place it is used is in this function

Reply:
`fd` does appear to be necessary. Otherwise the TAKE_FD call fails.